src/Statement/Grouper/StatementGrouper.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Statement\Grouper;
+
+use Wikibase\DataModel\Statement\StatementList;
+
+/**
+ * @since 3.2
+ *
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ */
+interface StatementGrouper {
+
+	/**
+	 * @param StatementList $statements
+	 *
+	 * @return StatementList[] An associative array, mapping statement group identifiers to
```

Comment:
Is there something these identifiers have in common? It's still not clear to me what as a user of this interface I can expect. It's possible they do not have anything in common, and that if I get passed an arbitrary implementation of the interface, I can make no assumptions about these keys. If that is the case, it is not bad, though then naming the concept might be confusing, as it somewhat implies it has relevant attributes to the user of the interface.

> This is an interface.

![obvious](https://cloud.githubusercontent.com/assets/146040/11332024/0aa10906-91c2-11e5-92f5-8ccc6e7fcae3.jpg)